### PR TITLE
feat: Natively support runner pre/post job hooks

### PR DESCRIPTION
**Pull request by winwinashwin**
Adds two optional string arguments to the module
- `runner_hook_job_started`
- `runner_hook_job_completed`

If not empty, registers the hook in `user-data.sh`

Ref: https://github.com/actions/runner/blob/main/docs/adrs/1751-runner-job-hooks.md

Closes: https://github.com/philips-labs/terraform-aws-github-runner/issues/4260
Discussion: https://github.com/philips-labs/terraform-aws-github-runner/issues/3854

### CODE REVIEW

**Comment by winwinashwin** [comment]
@npalm Could you please review?

**Review by npalm — COMMENTED** [review]
Nice contribution! I will check the PR in detail later. But could you check if you can enrich the multi-runner example with the pre / post hook. I would suggest to use one of the amazon linux examples.

**Comment by winwinashwin** [comment]
> Enrich the multi-runner example with the pre / post hook.

Thanks. Added examples in the PR.

**Comment by npalm** [comment]
> @npalm Anything I can help with to get this merged? Thanks for your time.

Not much, clear my calendar. I started last week from the bottom of the PR list. Do my best to check the PR this week.

File: modules/multi-runner/variables.tf [hunk 1]
```diff
@@ -78,6 +78,8 @@ variable "multi_runner_config" {
       cloudwatch_config                       = optional(string, null)
       userdata_pre_install                    = optional(string, "")
       userdata_post_install                   = optional(string, "")
+      runner_hook_job_started                 = optional(string, "")
```
Comment (npalm) [inline modules/multi-runner/variables.tf:81]: can you update the documentation section for this variable please?

File: examples/multi-runner/templates/runner-configs/linux-x64-ubuntu.yaml [hunk 2]
```diff
@@ -52,3 +52,12 @@ runner_config:
       prefix_log_group: true
       file_path: /opt/actions-runner/_diag/Runner_**.log
       log_stream_name: "{instance_id}/runner"
+  runner_hook_job_started: |
```
Comment (npalm) [inline examples/multi-runner/templates/runner-configs/linux-x64-ubuntu.yaml:55]: can you remove the pre post from this example. This to have one example (linux) based without the pre post. With the two other we cover the standard (amazonn linux() as well ubuntu via custom user_data.

File: modules/multi-runner/variables.tf (hunk 1 above)
Comment (winwinashwin) [reply modules/multi-runner/variables.tf:81]: Interesting. I was unaware that this (documentation) needs to be explicitly handled as for the other submodules docs are autogenerated.

Just a thought - If the plan is to move to multi-runner by default, then maybe we should think about making this more clear. Either we should think through autogenerating docs or make it clear to contributors that documentation needs to be handled manually.

**Comment by winwinashwin** [comment]
@npalm Thank you for your review. Updated changes as suggested.

**Comment by winwinashwin** [comment]
@npalm Anything pending here?

**Comment by npalm** [comment]
pushed a change to simply the test script in examples, now it works

image

File: modules/multi-runner/variables.tf (hunk 1 above)
Comment (npalm) [reply modules/multi-runner/variables.tf:81]: You right, most of the docs is generated. Let improve the docs in a seperate PR. We moving the repo first before moving modules around.

File: examples/multi-runner/templates/runner-configs/linux-x64.yaml [hunk 3]
```diff
@@ -29,3 +29,12 @@ runner_config:
...
     delay_in_seconds: 180
+  runner_hook_job_started: |
+    echo "Running pre job hook as \$(whoami)"
+
+    # Clean github workspace from previous runs
+    if [[ -n "\$GITHUB_WORKSPACE" ]]; do
+      rm -rf "\$GITHUB_WORKSPACE"
+    done
+  runner_hook_job_completed: |
+    echo "Running post job hook as \$(whoami)"
```
[the remark this answers is not in the document]
Comment (npalm) [reply examples/multi-runner/templates/runner-configs/linux-x64.yaml]: I have simplified the script, by the way if and do are a mismatch :)

**Review by npalm — APPROVED** [review]

File: examples/multi-runner/templates/runner-configs/linux-x64.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (winwinashwin) [reply examples/multi-runner/templates/runner-configs/linux-x64.yaml]: Oops, lost my presence of mind a little over there. Thanks for the changes!

**Pull request merged**